src/InfoViz/Core/SelectionProvider/histogram2d.js

```diff
@@ -2,19 +2,128 @@
 // Histogram2d
 // ----------------------------------------------------------------------------
 //
+// ===> SET
+//
 //  const payload = {
 //    type: 'histogram2d',
 //    data: {
 //      x: 'temperature',
 //      y: 'pressure',
-//      bins: [...]
-//    }
+//      bins: [...],
+//      score: 0,
+//      annotationId: 24, // Generation of the annotation
+//      selectionId: 23, // Generation of the selection
+//    },
 //  }
 //
+// ===> GET
+//
 //  const query = {
 //    type: 'histogram2d',
-//    axes: ['temperature', 'pressure']
+//    axes: ['temperature', 'pressure'],
+//    score: [0, 2],
 //  }
+//
+// const response = [
```

Comment:
Ah, OK, I didn't think it was an already-understood format. Fine with me.

On Wed, Aug 3, 2016 at 1:40 PM, Sebastien Jourdain <notifications@github.com
> wrote:

> In src/InfoViz/Core/SelectionProvider/histogram2d.js
> <https://github.com/Kitware/paraviewweb/pull/181#discussion_r73377146>:
>
> >  //  }
> > +//
> > +// const response = [
>
> The reason of that duplication (x, y, annotationId, selectionId) is that
> each item in the response is a self contained histogram2d instead of adding
> a data conversion at the store time and the usage time. And the lot of
> duplication is somewhat arguable ;-).
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kitware/paraviewweb/pull/181/files/3a1ddaf4cd7ac446b6eef25216c9a36a89801403#r73377146>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AS_pne89JPxknFqlf7v7apz-EJEEeh7Iks5qcNJ2gaJpZM4JbKHl>
> .
>